core/src/main/java/org/infinispan/interceptors/impl/CacheMgmtInterceptor.java

```diff
@@ -689,6 +689,8 @@ public long getAverageRemoveTimeNanos() {
          displayName = "Approximate number of entries"
    )
    public long getApproximateEntries() {
+      if (!globalConfiguration.metrics().accurateSize()) return -1;
```

Comment:
This change doesn't look correct to me. The accurate size setting is not for this method but so that this method is used instead of actual size as this method should be a constant time operation or as minimal as possible to get a worthwhile estimation.

Reply:
My guess is there may be a disconnect from what the docs say for the stats.

Reply:
my 2 cents, the approximate* attributes should never return `-1`.

Reply:
Exactly, which is why this looked wrong to me.